scripts/download.pl: Remove stale download site definitions

hnyman opened the pull request:
I did the first round yesterday when I noticed problem with GNOME, but decided to check also all other mirror site aliases. And there were several dead or erroneous sites...

-----
Remove the stale site definitions from APACHE, GNU, KERNEL etc.

* Remove site that had dropped APACHE
* Convert non-responding ftp sites to https/http
* Remove KERNEL site leading to wrong directory
* Remove dead sites

BKPepe commented on scripts/download.pl:
```diff
@@ -277,21 +276,18 @@ sub cleanup
 		push @mirrors, "http://ftp.kddilabs.jp/GNU/gnu/$1";
 		push @mirrors, "http://www.nic.funet.fi/pub/gnu/gnu/$1";
-		push @mirrors, "http://mirror.internode.on.net/pub/gnu/$1";
 		push @mirrors, "http://mirror.navercorp.com/gnu/$1";
```
I think we can here use HTTPS instead of HTTP, right?

hnyman replied:
Not all of those sites do support https, so even some of my fixes were just http.

But I could do a second evaluation, and try to switch as many http sites to https, if that is preferred.

BKPepe replied:
From my side, we should really switch those websites from HTTP to HTTPS, if they support it because it increases security and also would be a good addition to the already merged commit - https://github.com/openwrt/openwrt/commit/90c6e3aedf167b0ae1baf376e7800a631681e69a

hnyman commented:
I added a second commit, where I converted http URLs to https, if possible.
A few sites did not support https or did give certificate errors, so I left them as http.

(I can naturally squash this if requested, but for clarity I left them as two commits by now.)

hauke commented on scripts/download.pl:
```diff
@@ -274,17 +274,17 @@ sub cleanup
 		push @mirrors, "https://mirror.netcologne.de/savannah/$1";
 		push @mirrors, "https://mirror.csclub.uwaterloo.ca/nongnu/$1";
-		push @mirrors, "http://ftp.acc.umu.se/mirror/gnu.org/savannah/$1";
+		push @mirrors, "https://ftp.acc.umu.se/mirror/gnu.org/savannah/$1";
```
This mirror is listed 2 times

hnyman replied:
duplicate removed.  (had been both http and ftp, thus the duplicate)

hauke commented:
Which kernel mirror was leading to the wrong directory?

xback commented:
I also noticed [This](https://git.openwrt.org/?p=openwrt/staging/xback.git;a=commit;h=ac37ab6bcaeec3d3af92155055a8a428f8132b6c) 

Download was very slow at rougly 1 .. 2 mbit (from Belgium at least)
Switching to this new source improved gcc download a lot.

No idea if you encounter the same?

(Feel free to use this commit too)

hnyman committed:
scripts/download.pl: Remove stale download sites and convert to https

Remove the stale site definitions from @APACHE, KERNEL etc.

* Remove site that had dropped APACHE
* Remove KERNEL site leading to wrong directory
* Remove dead sites
* Convert ftp/http URLs to https, if possible. Remove duplicate

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>

hnyman commented:
> Which kernel mirror was leading to the wrong directory?

https://mirrors.tuna.tsinghua.edu.cn/kernel/ led to quite other contents than all other kernel download links.


I removed the one duplicate and squashed the two commits.
